src/grammar/query.rs

```diff
@@ -988,6 +1507,170 @@ Root@0..61
         );
     }
 
+    #[test]
+    fn test_cte_procedure() {
+        check(
+            parse(
+                "WITH PROCEDURE print_contact(
+    in_customer_id NUMBER 
+)
+IS
+  r_contact contacts%ROWTYPE;
+BEGIN
+  -- get contact based on customer id
+  SELECT *
+  INTO r_contact
+  FROM contacts
+  WHERE customer_id = p_customer_id;
+
+  -- print out contact's information
+  dbms_output.put_line( r_contact.first_name || ' ' ||
+  r_contact.last_name || '<' || r_contact.email ||'>' );
+
+END; 
```

Comment:
Lets cut down the test data to what we actually want to test. The details of the procedure are irrelevant for us here.
```suggestion
BEGIN
  NULL;
END; 
```